duckduckgo/macos-browser
# Appcast management tool improvements

Task/Issue URL: https://app.asana.com/0/1203301625297703/1205739517230527/f
https://app.asana.com/0/1203301625297703/1205739517230524/f
https://app.asana.com/0/1203301625297703/1205528551047876/f

**Description**:
Improvements of our appcast management script to sure sure developers use correct version of Sparkle binary tools, have correct signing keys installed and that all delta files are ready for s3 upload in `sparkle-updates` directory.

**Steps to test this PR**:
Create a random release notes file in `~/Downloads/release_notes`
**Test Sparkle binary tools version check**
1. Install old Sparkle tools (https://github.com/sparkle-project/Sparkle/releases)
2. Execute the script
`./scripts/appcast_manager/appcastManager.swift --release-to-public-channel --version 1.61.0 --release-notes ~/Downloads/release_notes`
3. Verify the script ends with error code 1 and appropriate error message
4. Install latest Sparkle tools
5. Verify the script finishes with success

**Test signing keys check**
1. Install correct signing keys (you probably have them installed)
3. Execute the script 
`./scripts/appcast_manager/appcastManager.swift --release-to-public-channel --version 1.61.0 --release-notes ~/Downloads/release_notes`
6. Make sure it finishes without any issues
3. Modify the public key in the script (See `desiredPublicKey`)
7. Execute the script and verify it exits with error.
`./scripts/appcast_manager/appcastManager.swift --release-to-public-channel --version 1.61.0 --release-notes ~/Downloads/release_notes`

**Test presence of all delta files in `sparkle-updates` directory**
1. 2. Execute the script 
`./scripts/appcast_manager/appcastManager.swift --release-to-public-channel --version 1.61.0 --release-notes ~/Downloads/release_notes`
3. Verify that `sparkle-updates/old_updates` is empty.



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
